api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -354,41 +362,26 @@ private DbConceptSet getDbConceptSet(
         workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
             workspaceNamespace, workspaceId, workspaceAccessLevel);
 
-    DbConceptSet conceptSet = conceptSetService.findOne(conceptSetId);
-    if (conceptSet == null || workspace.getWorkspaceId() != conceptSet.getWorkspaceId()) {
-      throw new NotFoundException(
-          String.format(
-              "No concept set with ID %s in workspace %s.",
-              conceptSetId, workspace.getFirecloudName()));
-    }
+    DbConceptSet conceptSet =
+        conceptSetService
+            .findOne(conceptSetId, workspace.getWorkspaceId())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        String.format(
+                            "No concept set with ID %s in workspace %s.",
+                            conceptSetId, workspace.getFirecloudName())));
     return conceptSet;
   }
 
-  public static ConceptSet toClientConceptSet(DbConceptSet dbConceptSet) {
-    return new ConceptSet()
-        .etag(Etags.fromVersion(dbConceptSet.getVersion()))
-        .lastModifiedTime(dbConceptSet.getLastModifiedTime().getTime())
-        .creationTime(dbConceptSet.getCreationTime().getTime())
-        .description(dbConceptSet.getDescription())
-        .id(dbConceptSet.getConceptSetId())
-        .name(dbConceptSet.getName())
-        .domain(dbConceptSet.getDomainEnum())
-        .participantCount(dbConceptSet.getParticipantCount())
-        .survey(dbConceptSet.getSurveysEnum())
-        .creator(
-            dbConceptSet.getCreator() != null ? dbConceptSet.getCreator().getUsername() : null);
-  }
-
   private ConceptSet toClientConceptSetWithConcepts(DbConceptSet dbConceptSet) {
-    ConceptSet conceptSet = toClientConceptSet(dbConceptSet);
-    if (!dbConceptSet.getConceptIds().isEmpty()) {
-      Iterable<DbConcept> concepts = conceptService.findAll(dbConceptSet.getConceptIds());
-      conceptSet.setConcepts(
-          Streams.stream(concepts)
-              .map(ConceptsController::toClientConcept)
-              .sorted(CONCEPT_NAME_ORDERING)
-              .collect(Collectors.toList()));
-    }
+    ConceptSet conceptSet = conceptSetMapper.dbModelToClient(dbConceptSet);
+    Iterable<DbConcept> concepts = conceptService.findAll(dbConceptSet.getConceptIds());
+    conceptSet.setConcepts(
+        Streams.stream(concepts)
+            .map(ConceptsController::toClientConcept)
+            .sorted(CONCEPT_NAME_ORDERING)
+            .collect(Collectors.toList()));
```

Comment:
Removed the conditional because findAll return an empty collection of no concept Ids match.

Reply:
Yes, and you drop a `!` for the win.